docs/src/events.md

```diff
@@ -0,0 +1,168 @@
+---
+id: events
+title: "Events"
+---
+
+Playwright allows listening to various types of events happening in the browser, such
+as network requests, creation of child pages, dedicated workers etc. There are several
+ways to subscribe to such events:
+
+<!-- TOC -->
+
+## Waiting for event
+
+There is a number of methods which wait for a particular condition or an event happening
+in the page. They are usually the preferred choice. Here are a few examples:
+
+Waiting for a request with a specified url:
+
+```js
+const [request] = await Promise.all([
+  page.waitForRequest('**/*logo*.png'),
+  page.goto('https://wikipedia.org')
+]);
+console.log(request.url());
+```
+
+```java
+// The callback lambda defines scope of the code that is expected to
+// trigger request.
+Request request = page.waitForRequest("**/*logo*.png", () -> {
+  page.navigate("https://wikipedia.org");
+});
+System.out.println(request.url());
+```
+
+```python async
+async with page.expect_request("**/*logo*.png") as first:
+  await page.goto("https://wikipedia.org")
+first_request = await first.value
+print(first_request.url)
+```
+
+```python sync
+with page.expect_request("**/*logo*.png") as first:
+  page.goto("https://wikipedia.org")
+print(first.value.url)
+```
+
+Waiting for popup window:
+
+```js
+const [popup] = await Promise.all([
+  page.waitForEvent('popup'),
+  page.evaluate('window.open()')
+]);
+await popup.goto('https://wikipedia.org');
+```
+
+```java
+// The callback lambda defines scope of the code that is expected to
+// create popup window.
+Page popup = page.waitForPopup(() -> {
+  page.evaluate("window.open()");
+});
+popup.navigate("https://wikipedia.org");
+```
+
+```python async
+async with page.expect_popup() as popup:
+  await page.evaluate("window.open()")
+child_page = await popup.value
+await child_page.goto("https://wikipedia.org")
+```
+
+```python sync
+with page.expect_popup() as popup:
+  page.evaluate("window.open()")
+popup.value.goto("https://wikipedia.org")
+```
+
+## Adding event listener
+
```

Comment:
I'd add a small description to preface the code snippet, along these lines:

Playwright supports traditional language mechanisms to subscribe and unsubscribe from events

Reply:
Done.